### PR TITLE
tried to translate with deepl

### PR DESCRIPTION
**Pull request by Zarif99**

### CODE REVIEW

**Comment by NikBelyaev** [comment]
After all, create a classmethod `from_settings` for base translate engine, that receives the settings object and inits the engine.

```
class BaseEngine:
   SETTINGS_KEY = ''

@classmethod
def from_settings(cls, settings):
    engine_settings = getattr(settings, cls.SETTINGS_KEY)
    return cls(...)
```

**Comment by NikBelyaev** [comment]
There's also issues with templates and paths
let me find a nice solution and get back to you

File: src/blog_vi/translations/providers/deepl.py [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 
 
 class DeeplTranslateProvider(BaseTranslateProvider):
     id = 'deepl'
+    settings_key = 'deepl_translator'
 
     def translate(self, text: str, source_abbreviation: str, target_abbreviation: str) -> str:
-        translator = deepl.Translator(DEEPL_API_KEY)
+        api_key = getattr(self, '_BaseTranslateProvider__api_key')
+        translator = deepl.Translator(auth_key=api_key)
```
Comment (NikBelyaev) [inline src/blog_vi/translations/providers/deepl.py:11]: remove this, you have a `self.translator`

**Review by NikBelyaev — CHANGES_REQUESTED** [review]

File: src/blog_vi/translations/providers/deepl.py (hunk 1 above)
Comment (Zarif99) [reply src/blog_vi/translations/providers/deepl.py:11]: This is just instance of deepl.Translator() class, not identifying which translato

File: src/blog_vi/__main__.py [hunk 2]
```diff
@@ -325,8 +329,20 @@ def generate_blog(workdir: Path) -> None:
...
-    from .translations.engine import TranslateEngine
-    engine = TranslateEngine(index, 'en')
-    engine.translate()
-
     index.generate()
+
+    if settings.translate_articles:
+        try:
+            if settings.source_abbreviation is None:
+                print(f'[-] Please provide source language')
```
Comment (NikBelyaev) [inline src/blog_vi/__main__.py:337]: `print(f'[-] Please, provide a source language abbreviation.')`

File: src/blog_vi/__main__.py [hunk 3]
```diff
@@ -325,8 +329,20 @@ def generate_blog(workdir: Path) -> None:
...
+                sys.exit(1)
+            engine = TranslateEngine(index, settings.source_abbreviation)
+        except ProviderSettingsNotFound:
+            print('[-] Provider settings not found')
```
Comment (NikBelyaev) [inline src/blog_vi/__main__.py:341]: `print('[-] Settings for translate provider {settings.translator}')`

**Review by NikBelyaev — CHANGES_REQUESTED** [review]

File: src/blog_vi/translations/engine.py [hunk 4]
```diff
@@ -7,38 +9,66 @@ def __init__(self, landing: Landing, source_abbreviation: str):
...
 
         self.settings = landing.settings
+        self.translator = self.get_translate_engine(self.settings)
 
-        # TODO:
-        #   - Validate settings
-        #   - Get translate provider from registry
+    def get_translate_engine(self, settings):
+        translator_cls = translation_provider_registry.get_provider(settings.translator)
+        return translator_cls.from_settings(settings)
```
Comment (NikBelyaev) [inline src/blog_vi/translations/engine.py:16]: This should raise an error if engine not found

**Review by NikBelyaev — CHANGES_REQUESTED** [review]

File: src/blog_vi/translations/engine.py (hunk 4 above)
Comment (Zarif99) [reply src/blog_vi/translations/engine.py:16]: `get_provider` has try except and it raise error if engine not found, or you mean not it

Comment (Zarif99) [reply src/blog_vi/translations/engine.py:16]: `translator_cls` can be DeeplTranslator/GoogleTranslator or `get_provider` function raise error

Comment (NikBelyaev) [reply src/blog_vi/translations/engine.py:16]: get provider shouldn't raise exceptions, as its context is not about it. This method should raise them

Comment (NikBelyaev) [reply src/blog_vi/translations/engine.py:16]: if get provider is none -> raise an error

File: src/blog_vi/templates/blog.html [hunk 5]
```diff
@@ -12,9 +12,9 @@
           integrity="sha512-iBBXm8fW90+nuLcSKlbmrPcLa0OT92xO1BIsZ+ywDWZCvqsWgccV3gFoRBv0z+8dLJgyAHIhR35VZc2oM/gI1w=="
           crossorigin="anonymous"/>
     <link href="https://fonts.googleapis.com/css?family=Nunito:400,700,800" rel="stylesheet">
-    <link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
-    <link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
-    <link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
+    <link rel="apple-touch-icon" sizes="180x180" href="../../apple-touch-icon.png">
+    <link rel="icon" type="image/png" sizes="32x32" href="../../favicon-32x32.png">
+    <link rel="icon" type="image/png" sizes="16x16" href="../../favicon-16x16.png">
```
Comment (NikBelyaev) [inline src/blog_vi/templates/blog.html:17]: Change all of these to "/{{ settings.blog_root }}/favicon..."

**Review by NikBelyaev — CHANGES_REQUESTED** [review]

**Pull request merged**